yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -207,6 +211,17 @@ private[yarn] class YarnAllocator(
   }
 
   /**
+   * Gets the executor loss reason for a disconnected executor.
+   * Note that this method is expected to be called exactly once per executor ID.
+   */
+  def getExecutorLossReason(executorId: String): ExecutorLossReason = synchronized {
+    allocateResources()
```

Comment:
More just that I wanted to be absolutely certain that the processCompletedContainers() call was giving the most up to date state. Other threads asynchronously updating the completed container state might not have given the most up to date information at the point where this method was called.